CMSgov/mint-app
# EASI-1465/craco-uswds-upgrade

# EASI-1465

## Changes and Description

- Upgraded to USWDS 3.3.0
- Upgraded to CRACO 7.0
- Upgraded react-scripts (CRA)
- Upgraded a few other packages for compatibility
- Removed craco-sass-resource-loader
- Removed old uswds theme package
- Adjust craco.config.js to compile new uswds 3.0
- Updated style entry point (index.scss) to import only used packages for better compile
- Imported uswds core in all stylesheet that utilized themes, functions, mixins, etc.
- Deprecated uswdsUtilities.scss
- Replaced EASI' `UsGovBanner` with Truss' `GovBanner`

No snapshots needed to be updated!

## Outstanding work:
Update Truss package (Will be in another ticket branch at this point)

I was decided to PoC on MINT for familiarity, but this upgrade can easily be adapted on EASI.


## How to test this change

- Make sure all tests run successfully, and try to spot any stylistic differences.  (Should be none)


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
